cli/flox-rust-sdk/src/models/environment/managed_environment.rs

```diff
@@ -577,10 +573,20 @@ impl ManagedEnvironment {
 
 /// Utility instance methods
 impl ManagedEnvironment {
+    /// Lock the environment to the current revision
+    fn lock_pointer(&self) -> Result<(), ManagedEnvironmentError> {
+        write_pointer_lockfile(
+            self.path.join(GENERATION_LOCK_FILENAME),
+            &self.floxmeta,
+            remote_branch_name(&self.system, &self.pointer),
```

Comment:
I just realized this could bump `rev` when we don't mean to - if I pull an environment at rev A, somebody pushes rev B, I do an operation with `-r` that fetches the environment, and then I make a change that takes me from rev A to rev C, my lock will set `rev` to B. I think(?) that's undesirable, and rev should always be in local_rev's history. No need to fix in this PR, but maybe we could at least add a TODO?

Reply:
added a todo